File: 3254.py
# 3254. Find the Power of K-Size Subarrays I
from collections import List

# class Solution:
#     def resultsArray(self, nums: List[int], k: int) -> List[int]:
#         l = len(nums)
#         result = [-1] * (l - k + 1)

#         for start in range(len - k):
#             isSorted = True

#             for i in range(start, start + k - 1):
#                 if (nums[i+1] != nums[i] + 1):
#                     isSorted = False
#                     break

#             if isSorted:
#                 result[start] = nums[start + k - 1]
        
#         return result

# Optimal Using Counter
class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        if k == 1:
            return nums
        
        l = len(nums)
        result = [-1] * (l - k + 1)
        consecutive_count = 1

        for start in range(l - 1):
            if nums[start] + 1 == nums[start + 1]:
                consecutive_count += 1
            else:
                consecutive_count = 1

            if consecutive_count >= k:
                result[start - k + 2] = nums[start + 1]

        return result; 

File: 3254.cpp
// 3254. Find the Power of K-Size Subarrays I
#include <bits/stdc++.h>
using namespace std;

class Solution{
public:
    vector<int> resultsArray(vector<int>& nums, int k){
        int len = nums.size();
        vector<int> result(len - k + 1);

        for (int start = 0; start <= len - k; start++){
            bool isSorted = true;

            for (int i = start; i < start + k - 1; i++){
                if (nums[i + 1] != nums[i] + 1){
                    isSorted = false;
                    break;
                }
            }

            if (isSorted){
                result[start] = nums[start + k - 1];
            } else {
                result[start] = -1;
            }
        }

        return result;
    }
};

int main(){

    return 0;
}